Objects/memoryobject.c

```diff
@@ -1365,6 +1367,13 @@ memory_cast(PyMemoryViewObject *self, PyObject *args, PyObject *kwds)
             "memoryview: cannot cast view with zeros in shape or strides");
         return NULL;
     }
+
+    if (readonly && !PyBool_Check(readonly)){
```

Comment:
Instead of this, just use the `p` type format in `PyArgs_ParseTuple` above (see https://docs.python.org/3/c-api/arg.html#other-objects).